src/http/cookie.cr

```diff
@@ -50,7 +50,7 @@ module HTTP
     end
 
     def to_cookie_header
-      "#{@name}=#{URI.escape value}"
+      "#{URI.encode_www_form(@name)}=#{URI.encode_www_form(value)}"
```

Comment:
optional optimization: stream to the IO instead of creating many intermediate strings.